sorting/selection_sort_recursive.c

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file
+ * @author [Dhruv Pasricha](https://github.com/DhruvPasricha)
+ * @brief [Selection Sort](https://en.wikipedia.org/wiki/Selection_sort)
+ * implementation using recursion.
+ */
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
```

Comment:
Please add a one-line comment to tell what the library/header is for (see the example below).
```c++
#include <cassert>   /// for assert
#include <iostream>  /// for IO operations
```

Reply:
I have done all the required changes.😄

Reply:
I have made the required changes. 😄